packages/git/src/browser/git-scm-provider.ts

```diff
@@ -195,6 +195,9 @@ export class GitScmProvider implements ScmProvider {
 
     getUriToOpen(change: GitFileChange): URI {
         const changeUri: URI = new URI(change.uri);
+        if (change.status === GitFileStatus.Deleted) {
+            return changeUri.withScheme(GIT_RESOURCE_SCHEME).withQuery('HEAD');
+        }
```

Comment:
I don't think this code is complete.  Consider what happens if you modify a file, stage the file, then delete the file.  You should see the file with 'M' status in the staged section and with a 'D' status in the unstaged section.  (If one actually tries this out then you will see an incorrect 'M' in the unstaged section.  That is because of this bug https://github.com/eclipse-theia/theia/issues/7789).  If you click anything in the unstaged section then you should see the diff between the staged and the unstaged.  That means if you click on the deleted file in the unstaged section then you should see the contents of the file in the index, not the contents of the file in HEAD.

Until https://github.com/eclipse-theia/theia/issues/7789 is fixed you won't be able to test all the cases here.  So on that basis I would be happy with the changes currently in git-scm-provider with a TODO that more work is needed when https://github.com/eclipse-theia/theia/issues/7789 is fixed.  I think the suggestion will be correct in all cases, but I can't test it without fixing 7789.

```suggestion
        if (change.status === GitFileStatus.Deleted) {
           if (change.staged) { 
               return changeUri.withScheme(GIT_RESOURCE_SCHEME).withQuery('HEAD');
           } else {    
               return changeUri.withScheme(GIT_RESOURCE_SCHEME);
            }
         }
```